format/FlightSql.proto

```diff
@@ -1778,6 +1789,44 @@ message CommandPreparedStatementUpdate {
   bytes prepared_statement_handle = 1;
 }
 
+  /*
+   * Represents a bulk ingestion request. Used in the command member of FlightDescriptor
+   * for the the RPC call DoPut to cause the server load the contents of the stream's
+   * FlightData into the target destination.
+   */
+message CommandStatementIngest {
+  option (experimental) = true;
+
+  // Describes the behavior for loading bulk data.
+  enum IngestMode {
+    // Ingestion behavior unspecified.
+    INGEST_MODE_UNSPECIFIED = 0;
+    // Create the target table. Fail if the target table already exists.
+    INGEST_MODE_CREATE = 1;
+    // Append to an existing target table. Fail if the target table does not exist.
+    INGEST_MODE_APPEND = 2;
+    // Drop the target table if it exists. Then follow INGEST_MODE_CREATE behavior.
+    INGEST_MODE_REPLACE = 3;
+    // Create the target table if it does not exist. Then follow INGEST_MODE_APPEND behavior.
+    INGEST_MODE_CREATE_APPEND = 4;
+  }
+
+  // The ingestion behavior.
+  IngestMode mode = 1;
+  // The table to load data into.
+  string target_table = 2;
+  // The db_schema of the target_table to load data into. If unset, a backend-specific default may be used.
+  optional string target_schema = 3;
+  // The catalog of the target_table to load data into. If unset, a backend-specific default may be used.
+  optional string target_catalog = 4;
+  // Use a temporary table for target_table.
+  optional bool temporary = 5;
+  // Perform the ingestion as part of this transaction (if unset, the query is auto-committed).
+  optional bool transaction_id = 6;
+  // Backend-specific options.
+  map<string, string> options = 1000;
```

Comment:
Why do we want to use `1000` instead of `7` here?

Reply:
The idea is that `options` can be used for any parameters not otherwise supported by the spec, which makes it different from any one parameter. By giving this a high number, it leaves room for more explicit parameters to be added to the spec at lower numbers as the spec evolves.

Reply:
OK. Could you add a comment how to add a new explicit parameter?

Reply:
Yes, I've just added a comment.

Reply:
I'm not sure I understand this rationale. What would it change to add new parameters at higher numbers, rather than at lower numbers?